% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
  \setmainfont[]{Helvetica Neue}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\usepackage[normalem]{ulem}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={DoE: Design of Experiment: RNA-seq Library Prep \& Target Enrichment (TE) Panel},
  pdfauthor={Patrick Cherry},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{DoE: Design of Experiment: RNA-seq Library Prep \& Target
Enrichment (TE) Panel}
\author{Patrick Cherry}
\date{2023-05-24}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\hypertarget{introduction-and-background}{%
\subsection{Introduction and
Background}\label{introduction-and-background}}

I had the idea that, given the success Panel A's bioinformatic
performance, \emph{it could be useful to show that TE panels work well
for an extension of said bioinformatic performance.}

To do so, I propose using multiplexed capture, 100 ng (and optionally 10
ng of RNA input), and technical replicates (3, perhaps 2). These are
parameterized in this DoE script below and the resulting sample plan is
exported to google sheets.

\hypertarget{procedure}{%
\subsection{Procedure}\label{procedure}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{file\_pref }\OtherTok{\textless{}{-}} \StringTok{"2023\_05\_24\_RNA\_TE\_\_sensitivty\_DoE"}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{panel\_info }\OtherTok{\textless{}{-}} \FunctionTok{tribble}\NormalTok{(}
  \SpecialCharTok{\textasciitilde{}}\NormalTok{panel,  }\SpecialCharTok{\textasciitilde{}}\NormalTok{panel\_size,  }\SpecialCharTok{\textasciitilde{}}\NormalTok{needed\_sequencing,}
  \StringTok{"TE Panel A"}\NormalTok{, }\FloatTok{3.0}\NormalTok{, }\ConstantTok{NA}\NormalTok{,}
  \StringTok{"TE Panel B"}\NormalTok{,  }\FloatTok{36.8}\NormalTok{, }\ConstantTok{NA}\NormalTok{,}
  \StringTok{"TE Panel C"}\NormalTok{,  }\FloatTok{35.8}\NormalTok{, }\ConstantTok{NA}\NormalTok{,}
  \StringTok{"Whole Transcriptome"}\NormalTok{, }\ConstantTok{NA}\NormalTok{, }\ConstantTok{NA}\NormalTok{,}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{doe-with-blocking-for-multiple-operators}{%
\subsection{DoE with blocking for multiple
operators}\label{doe-with-blocking-for-multiple-operators}}

I will block for the operators carrying out library prep, because
operator is a known source of variation that is not relevant to
understanding the effect of TE panel, RNA input mass, or concentration
on performance.

Blocking is the non-random assignment of samples to groups to minimize
differences in the sample composition between the groups such that any
effect of the grouping can be determined by the model and ignored
(modeled out quantitatively and precisely).

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_sensitivity\_doe}\SpecialCharTok{$}\NormalTok{D;}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.3789291
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_sensitivity\_doe}\SpecialCharTok{$}\NormalTok{diagonality}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.871
\end{verbatim}

Diagonality is the degree to which the blocked variables are
uncorrelated: a diagonality of 1.0 is perfectly uncorrelated. A value of
0.871 is moderate. We are getting values less than 1.0, because not
every number of unique sample ( 2 * 4 * \sout{5} ) factors to be
processed is divisible by the number of blocking groups. We will see
this effect illustrated in the ``Check orthogonality of blocking''
section.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_blocking\_df }\OtherTok{\textless{}{-}} \FunctionTok{bind\_rows}\NormalTok{(}
 \FunctionTok{mutate}\NormalTok{(rna\_TE\_\_sensitivity\_doe}\SpecialCharTok{$}\NormalTok{Blocks}\SpecialCharTok{$}\NormalTok{B1, }\StringTok{"operator"} \OtherTok{=} \StringTok{"Operator A"}\NormalTok{),}
 \FunctionTok{mutate}\NormalTok{(rna\_TE\_\_sensitivity\_doe}\SpecialCharTok{$}\NormalTok{Blocks}\SpecialCharTok{$}\NormalTok{B2, }\StringTok{"operator"} \OtherTok{=} \StringTok{"Operator B"}\NormalTok{),}
\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(panel, conc, mass\_input)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{(panels\_to\_join }\OtherTok{\textless{}{-}}\NormalTok{ rna\_TE\_\_blocking\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{distinct}\NormalTok{(panel) }\SpecialCharTok{\%\textgreater{}\%}
   \FunctionTok{bind\_cols}\NormalTok{(}\FunctionTok{rename}\NormalTok{(panel\_info, }\StringTok{"panel\_name"} \OtherTok{=} \DecValTok{1}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}llrl@{}}
\toprule()
panel & panel\_name & panel\_size & needed\_sequencing \\
\midrule()
\endhead
1 & TE Panel A & 3.0 & NA \\
2 & TE Panel B & 36.8 & NA \\
3 & TE Panel C & 35.8 & NA \\
4 & Whole Transcriptome & NA & NA \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{(fusconcs\_to\_join }\OtherTok{\textless{}{-}}\NormalTok{ rna\_TE\_\_blocking\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{distinct}\NormalTok{(conc) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{bind\_cols}\NormalTok{(concentrations) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{rename}\NormalTok{(}\StringTok{"concentrations"} \OtherTok{=} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}rl@{}}
\toprule()
conc & concentrations \\
\midrule()
\endhead
-2 & 0.027 \\
-1 & 0.0027 \\
0 & 0.00027 \\
1 & 2.7e-05 \\
2 & 2.7e-06 \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{(massinput\_to\_join }\OtherTok{\textless{}{-}}\NormalTok{ rna\_TE\_\_blocking\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{distinct}\NormalTok{(mass\_input) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{bind\_cols}\NormalTok{(mass\_inputs) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{rename}\NormalTok{(}\StringTok{"mass\_inputs"} \OtherTok{=} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}rl@{}}
\toprule()
mass\_input & mass\_inputs \\
\midrule()
\endhead
-1 & 10 \\
1 & 100 \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_doe\_blocked }\OtherTok{\textless{}{-}}\NormalTok{ rna\_TE\_\_blocking\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(panels\_to\_join, }\AttributeTok{by =} \StringTok{"panel"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(fusconcs\_to\_join, }\AttributeTok{by =} \StringTok{"conc"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(massinput\_to\_join, }\AttributeTok{by =} \StringTok{"mass\_input"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{select}\NormalTok{(}\StringTok{"panel"} \OtherTok{=} \StringTok{"panel\_name"}\NormalTok{, }\StringTok{"conc"} \OtherTok{=} \StringTok{"concentrations"}\NormalTok{,}
         \StringTok{"mass\_input"} \OtherTok{=} \StringTok{"mass\_inputs"}\NormalTok{, }\StringTok{"operator"}\NormalTok{, panel\_size, needed\_sequencing) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(panel, }\FunctionTok{desc}\NormalTok{(conc), mass\_input) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\StringTok{"replicate\_num"} \OtherTok{=} \FunctionTok{row\_number}\NormalTok{(), }\AttributeTok{.by =} \FunctionTok{c}\NormalTok{(panel, conc, mass\_input)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{relocate}\NormalTok{(replicate\_num, }\AttributeTok{.after =}\NormalTok{ operator) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{rename}\NormalTok{(}\StringTok{"LP\_operator"} \OtherTok{=} \StringTok{"operator"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(panel, mass\_input) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\StringTok{"capture"} \OtherTok{=} \FunctionTok{ceiling}\NormalTok{(}\FunctionTok{row\_number}\NormalTok{()}\SpecialCharTok{/}\DecValTok{6}\NormalTok{) ) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{relocate}\NormalTok{(}\StringTok{"capture"}\NormalTok{, }\AttributeTok{.after =}\NormalTok{ replicate\_num) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(panel, }\FunctionTok{desc}\NormalTok{(conc), mass\_input)}
\FunctionTok{head}\NormalTok{(rna\_TE\_\_doe\_blocked, }\AttributeTok{n =} \DecValTok{10}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1196}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.0761}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1196}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1304}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1522}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.0870}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1196}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 14\tabcolsep) * \real{0.1957}}@{}}
\toprule()
\begin{minipage}[b]{\linewidth}\raggedright
panel
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
conc
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
mass\_input
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
LP\_operator
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedleft
replicate\_num
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedleft
capture
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedleft
panel\_size
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
needed\_sequencing
\end{minipage} \\
\midrule()
\endhead
TE Panel A & 0.027 & 10 & Operator A & 1 & 1 & 3 & NA \\
TE Panel A & 0.027 & 10 & Operator B & 2 & 1 & 3 & NA \\
TE Panel A & 0.027 & 10 & Operator B & 3 & 1 & 3 & NA \\
TE Panel A & 0.027 & 100 & Operator B & 1 & 3 & 3 & NA \\
TE Panel A & 0.027 & 100 & Operator B & 2 & 3 & 3 & NA \\
TE Panel A & 0.027 & 100 & Operator B & 3 & 3 & 3 & NA \\
TE Panel A & 0.0027 & 10 & Operator A & 1 & 1 & 3 & NA \\
TE Panel A & 0.0027 & 10 & Operator A & 2 & 1 & 3 & NA \\
TE Panel A & 0.0027 & 10 & Operator A & 3 & 1 & 3 & NA \\
TE Panel A & 0.0027 & 100 & Operator A & 1 & 4 & 3 & NA \\
\bottomrule()
\end{longtable}

\hypertarget{check-orthogonality-of-blocking}{%
\subsubsection{Check orthogonality of
blocking}\label{check-orthogonality-of-blocking}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_doe\_blocked }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{count}\NormalTok{(LP\_operator, mass\_input)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}llr@{}}
\toprule()
LP\_operator & mass\_input & n \\
\midrule()
\endhead
Operator A & 10 & 30 \\
Operator A & 100 & 30 \\
Operator B & 10 & 30 \\
Operator B & 100 & 30 \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_doe\_blocked }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{count}\NormalTok{(LP\_operator, panel)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}llr@{}}
\toprule()
LP\_operator & panel & n \\
\midrule()
\endhead
Operator A & TE Panel A & 15 \\
Operator A & TE Panel B & 15 \\
Operator A & TE Panel C & 15 \\
Operator A & Whole Transcriptome & 15 \\
Operator B & TE Panel A & 15 \\
Operator B & TE Panel B & 15 \\
Operator B & TE Panel C & 15 \\
Operator B & Whole Transcriptome & 15 \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_doe\_blocked }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{count}\NormalTok{(LP\_operator, conc)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}llr@{}}
\toprule()
LP\_operator & conc & n \\
\midrule()
\endhead
Operator A & 2.7e-06 & 13 \\
Operator A & 2.7e-05 & 12 \\
Operator A & 0.00027 & 9 \\
Operator A & 0.0027 & 14 \\
Operator A & 0.027 & 12 \\
Operator B & 2.7e-06 & 11 \\
Operator B & 2.7e-05 & 12 \\
Operator B & 0.00027 & 15 \\
Operator B & 0.0027 & 10 \\
Operator B & 0.027 & 12 \\
\bottomrule()
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{googlesheets4}\SpecialCharTok{::}\FunctionTok{write\_sheet}\NormalTok{(rna\_TE\_\_doe\_blocked, }\AttributeTok{ss =} \StringTok{"sheet\_string\_goes\_here"}\NormalTok{,}
                           \AttributeTok{sheet =} \StringTok{"rna\_TE\_\_sensitivity\_doe"}\NormalTok{);}
\FunctionTok{write\_csv}\NormalTok{(rna\_TE\_\_doe\_blocked, }\FunctionTok{paste}\NormalTok{(}\StringTok{"doe\_design\_of\_experiment\_with\_library\_prep\_files/"}\NormalTok{,}
\NormalTok{                                           file\_pref,}
                                           \StringTok{"\_rna\_TE\_\_sensitivity\_doe.csv"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{analyze-captures}{%
\subsubsection{Analyze captures}\label{analyze-captures}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna\_TE\_\_doe\_blocked }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{count}\NormalTok{(capture, panel, mass\_input)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}rllr@{}}
\toprule()
capture & panel & mass\_input & n \\
\midrule()
\endhead
1 & TE Panel A & 10 & 6 \\
2 & TE Panel A & 10 & 6 \\
3 & TE Panel A & 10 & 3 \\
3 & TE Panel A & 100 & 3 \\
4 & TE Panel A & 100 & 6 \\
5 & TE Panel A & 100 & 6 \\
6 & TE Panel B & 10 & 6 \\
7 & TE Panel B & 10 & 6 \\
8 & TE Panel B & 10 & 3 \\
8 & TE Panel B & 100 & 3 \\
9 & TE Panel B & 100 & 6 \\
10 & TE Panel B & 100 & 6 \\
11 & TE Panel C & 10 & 6 \\
12 & TE Panel C & 10 & 6 \\
13 & TE Panel C & 10 & 3 \\
13 & TE Panel C & 100 & 3 \\
14 & TE Panel C & 100 & 6 \\
15 & TE Panel C & 100 & 6 \\
16 & Whole Transcriptome & 10 & 6 \\
17 & Whole Transcriptome & 10 & 6 \\
18 & Whole Transcriptome & 10 & 3 \\
18 & Whole Transcriptome & 100 & 3 \\
19 & Whole Transcriptome & 100 & 6 \\
20 & Whole Transcriptome & 100 & 6 \\
\bottomrule()
\end{longtable}

\hypertarget{conclusion}{%
\subsection{Conclusion}\label{conclusion}}

Great! We have an experiment design for testing the effect of these
panels against each other and against whole transcriptome sequencing.
The experiment had n = 3 replicates, and it is blocked for having two
operators carry out the RNA-seq library preps.

Importantly, when I get feedback on the design of this sample layout, I
can easily show my work. Even better, if changes are needed, the entire
design is programmed, and can be changed in seconds.

Let's go!

\end{document}
